stl/inc/xutility

```diff
@@ -2110,12 +2110,13 @@ namespace ranges {
 
             template <class _Ty>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                _STL_INTERNAL_CHECK(is_lvalue_reference_v<_Ty>);

```

Comment:
Isnt that a hard compile error if it isnt an lvalue reference? If that is the case why not static_assert?

Reply:
`_STL_INTERNAL_CHECK` is only engaged when running our own test harness, so it won't result in a hard compiler error for users.

Reply:
But if it is not an lvalue_reference it breaks anyway albeit with less reasonable error messages?

Reply:
I mean, maybe? It looks like everything here does `remove_reference_t` first which might admit rvalue references (but I have not dug into all the guts here yet). I was mostly explaining the mechanics of that `_STL_INTERNAL_CHECK` thing.

Reply:
I don't see any obvious reason non-lvalue-reference _Ty would otherwise break.

Reply:
We force all types to lvalue reference types in `operator()` before they get to `_Choice` or `_Choose` to reduce the total number of template specializations, e.g., `begin(span<int>{/*...*/})` and `begin(span_int_lvalue)` call `operator()<span<int>>` and `operator()<span<int&>` which both evaluate `_Choice<span<int>&>`. Previously, we performed  "lvalue forcing" only for the concept evaluations on lines 2116-2119, this PR does so closer to `operator()` for greater effect.

The `_STL_INTERNAL_CHECK` - together with exhaustive test coverage of `operator()` with all value categories of various types - ensures that we didn't screw up and write `_Choice<_Ty>` instead of `_Choice<_Ty&>` somewhere in `operator()`. A `static_assert` would be just as correct, but I don't want to pay a throughput cost (instantiate `is_lvalue_reference_v`) to validate the one function (`operator()`) that uses this machinery.